### PR TITLE
Update git to support updates of submodules

### PR DESCRIPTION
**Pull request by fredrike**
## Proposed change

Adds support for submodule updates in the add-on store.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/store/git.py [hunk 1]
```diff
@@ -156,6 +156,19 @@ async def pull(self):
...
+                        self.repo.git.submodule.root.update,
+                        {
+                            "recursive": False,
+                            "init": True,
+                            "force_remove": True,
+                            "force_reset": True,
+                        },
+                    )
+                )
+
```
Comment (pvizeli) [inline supervisor/store/git.py:171]: what is about shallow and depth?

Comment (fredrike) [reply supervisor/store/git.py:171]: Well, the `submodule.root.update` does now seem to support that. With **`"recursive": False`** I suppose that it implies `"depth": 1`.

https://gitpython.readthedocs.io/en/0.3/reference.html#git.objects.submodule.root.RootModule.update

Comment (pvizeli) [reply supervisor/store/git.py:171]: Okay. Do it not need `**` to expand it?

Comment (fredrike) [reply supervisor/store/git.py:171]: Not sure what you mean. But this suggestion update direct submodules.

Comment (pvizeli) [reply supervisor/store/git.py:171]: https://docs.python.org/3/library/functools.html#functools.partial . Did you test this code?

Comment (pvizeli) [reply supervisor/store/git.py:171]: just for the records, I'm fine with the change. I ask my self just if they would work like it is in that PR and how partial work

Comment (ludeeus) [reply supervisor/store/git.py:171]: it _might_ work, but we should be consistent, prefix the dict with `**`

**Review by pvizeli — APPROVED** [review]

**Pull request merged**